mitodl/open-discussions
# Added course catalog integration with micromasters

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2181 

#### What's this PR do?
Adds operations to pull in micromasters program data, which augments existing course/course run data from mitx.

#### How should this be manually tested?

- In micromasters, configure new `Course` and `CourseRun` records for a program in the RC environment (look at `/api/v0/catalog`) so it's configured against courses mitx is aware of. Double check that the courses are actually being imported. before you run the next step
- Run the `get_micromasters_data` task 
- Verify the `Program` now has `items` populated with `Course` references.

#### Any background context you want to provide?

Recent efforts to adapt the DRF serializers currently used in `api.py` proved either impossible or time consuming to get right. This PR introduces a new `etl` module that defines `extract` and `transform` methods that are specific to each provider (only micromasters for now) and a normalized set of loaders to persist the data. This is far simpler and can handle much more complex cases than DRF serializers were designed to address.
